lib/Dialect/MSFT/MSFTPasses.cpp

```diff
@@ -732,13 +732,16 @@ static ArrayAttr getGlobalRefs(Operation *op) {
 
 /// Helper to update GlobalRefOps after referenced ops bubble up.
 void PartitionPass::bubbleUpGlobalRefs(
-    Operation *op, llvm::DenseSet<hw::GlobalRefAttr> &refsMoved) {
+    Operation *op, StringAttr parentName,
+    llvm::DenseSet<hw::GlobalRefAttr> &refsMoved) {
   auto globalRefs = getGlobalRefs(op);
   if (!globalRefs)
     return;
 
   // GlobalRefs use the inner_sym attribute, so keep it up to date.
-  op->setAttr("inner_sym", StringAttr::get(op->getContext(), ::getOpName(op)));
+  auto oldInnerSym = op->getAttrOfType<StringAttr>("inner_sym");
+  auto newInnerSym = StringAttr::get(op->getContext(), ::getOpName(op));
+  op->setAttr("inner_sym", newInnerSym);
```

Comment:
Unrelated to this PR: we need to uniquify `inner_sym` -- we can't rely on `getOpName` to do this.